app/models/my_module.rb

```diff
@@ -500,4 +500,12 @@ def uncomplete
   def create_blank_protocol
     protocols << Protocol.new_blank_for_module(self)
   end
+
+  def coordinates
+    my_modules_on_same_position =
+      MyModule.where(experiment_id: experiment_id, x: x, y: y, archived: false).where.not(id: id)
+    if my_modules_on_same_position.any?
```

Comment:
May be just call directly:  experiment.my_modules.active.where(x: x, y: y).where.not(id: id).any?